### PR TITLE
Feature/feature branch

### PR DESCRIPTION
**Pull request by realisticattorney**
Milestones for this project were:

-Create a User model and a Post model

-Associate them in a one to many relation

-Render the author of post at index only if a user is logged in

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  </thead>
+
+  <tbody>
+    <% @posts.each do |post| %>
+      <tr>
+        <td><%= post.content %></td>
+        <% if user_signed_in? %>
+        <td> <%= post.user.username %> </td>
+        <% end %>
+           <% if user_signed_in? && post.user.username == current_user.username %>
```
Comment (mahmoud717) [inline app/views/posts/index.html.erb:20]: please move this logic code to the helpers

**Review by mahmoud717 — CHANGES_REQUESTED** [review]
### Changes required ♻️ 

**Hello @realisticattorney**

Great job with the project so far it looks amazing, but there are still some changes to be made:
 
---
### Requested 🔴 

- [ ] for the posts routes you are only allowed to have the routes that you will use. change the posts routes to `resources :posts, only: [:new, :create, :index]` 

- [ ] please move the logic code in the posts/index view to a helper.

 

---

please make these changes and request another code review.

if you have any questions don't hesitate to contact me on slack **MahmoudMohammad**

**Comment by realisticattorney** [comment]
The first correction is not valid. Posts need to be editable and removable by the author.

**Comment by realisticattorney** [comment]
The logic quoted is not repeated anywhere. Why would I define a method for it?  It's already using functions like user_signed_in that are self-descriptive. I don't see how that would make the code shorter nor more readable

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
# Status: Changes Required :vertical_traffic_light: :warning: 
## Hello @realisticattorney  :handshake: :handshake: :handshake: 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: I liked that you have tried to implemented the requested changes from the previous review, keep it up, and your project is really almost done.
#### :warning: However we need to make few changes to mark this project approved, don't feel discouraged because of being asked some changes more and more, that's beacuse this is a learning process, and we are here to help you to become an out-standing developer. See the requested change below :point_down: :point_down: :point_down: 
### Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes] :red_circle: 
 - [ ] You need to work on your commit messages, they need to be descriptive, don't just type whatever you want, like the dots, no you just need to add a descriptive message
- [ ] You have some logics in views: There shoud be absolutely NO logic in views. >Views are only for displaying information that became available to it from its corresponding controller not generating it. [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from `Pathwright`
Resources
    - [ ] View-related logic can be encapsulated in View helpers (app/helpers/*_helper.rb) => To help you understand what to change: Make sure that there are no if conditions, or loops in views
  
___
_If anything seems to be unclear to you, reachout and ask anything by commenting on this Pull Request and mention me as [descholar-ceo](https://github.com/descholar-ceo), I will receive a notification and will respond._

[GitHub](https://github.com/descholar-ceo) | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

_Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)_

**Comment by realisticattorney** [comment]
You're right. Thanks for the review

**Review by kevincappuccino — APPROVED** [review]
# [STATUS] PROJECT APPROVED 🍾

### I really like your project! Congrats @realisticattorney, you did an excellent job, you have my approval to go ahead and merge this pull request!


## As always, I wish you all the very best and see you both at the top🚀


👨🏻‍💻 **TSE: Kevin Alves**

  

📫 Have any question? Better call Kevin:

<a  href="mailto:kevinalves.js@gmail.com"> gmail</a> [Twitter Follow](https://twitter.com/intent/follow?screen_name=kevinnnmn) [](https://microverse-students.slack.com/team/UT86D3AP4)

**Pull request merged**